### PR TITLE
FY2019Q4 Quarterly refresh

### LINKED ISSUE #23: Calendar display is in UTC
I noticed my calendar appointments were all displayed in UTC. Might want to use .ToLocalTime() to mirror the user's calendar experience.

### PR DESCRIPTION
**Pull request by andrewconnell**
- Updated tutorial instructions
- Updated slide decks
- Updated project files
- Closes #23

### CODE REVIEW

**Comment by jasonjoh** [comment]
@andrewconnell this doesn't look right, the only file changed was the README. Did you forget to commit some other changes?

**Comment by andrewconnell** [comment]
@jasonjoh I'm not sure what happened as I thought everything was good on the last push, but apparently, it wasn't I just did another push and now I show a lot more changes... can you confirm?

**Comment by jasonjoh** [comment]
That looks more like what I expected ;)

**Comment by jasonjoh** [comment]
@andrewconnell @rob-windsor it looks like the projects were recreated for some reason. The port changed in the projects and a lot of the supporting packages are downgraded. What was the reason for that?

**Comment by andrewconnell** [comment]
@jasonjoh when we do quarterly refreshes, we run through the lab steps and recreate the project from scratch. We then replace the existing built solution in the demo folder with the results of our refresh. This way it's as current & clean with no historical artifacts in the project.

In doing these refreshes, we also update the default NuGet packages to the latest published stable versions. This is what consumers of the lab would experience... because these are all controlled by external services (NuGet, owners of packages, etc), its not really in our control.

I looked through the changed files and see what you're saying... looks like they are very small changes.

As for the port, that's dynamic for every project created in Visual Studio... in the past, we haven't updated it but if it's an issue for you, I can go back and do a quick fixup on those project settings.

**Comment by andrewconnell** [comment]
@jasonjoh Made a small change to the tutorial that was missed.

**Comment by jasonjoh** [comment]
@andrewconnell I didn't realize that was the procedure. So you're saying you do a New project, then update all of the NuGet packages that are installed by the template? If so, I'm wondering if we shouldn't include that as an explicit step in the tutorial. Or are you saying that you create the new project and accept the versions the template installs? That might explain the downgrades, it's possible I updated the packages at some point.

Regarding the port: It is a bit of a problem to change it. We use the completed project from this repository for our quick start experience, which registers the app and injects client ID/secret into the project. Currently the quick start tool is configured to register http://localhost:64107 as the redirect URI.

**Comment by rob-windsor** [comment]
@jasonjoh I can answer the project creation part of your question. I followed the steps in the first part of the tutorial exactly as written with one exception: I installed the current stable versions of Microsoft.Identity.Client and Microsoft.Graph rather than installing specific versions of those packages.  https://docs.microsoft.com/en-us/graph/tutorials/aspnet?tutorial-step=1

**Comment by jasonjoh** [comment]
Thanks @rob-windsor. We want to keep explicit versions there due to the tendency of MSAL to introduce breaking changes :). That allows the tutorial to continue to work even if we don't update it immediately.

**Comment by andrewconnell** [comment]
@jasonjoh We're on the same page there... I understand why you'd want the port not to chance, so we can change that and submit an additional commit. 

That's all that's needed correct? From this dialog, looks like everything else is good as-is.

**Comment by jasonjoh** [comment]
Thanks @andrewconnell let's update the port and add back explicit versions for MSAL/Graph in the instructions and we should be good.

File: Demos/01-create-app/graph-tutorial/packages.config [hunk 1]
```diff
@@ -4,21 +4,21 @@
   <package id="bootstrap" version="4.3.1" targetFramework="net472" />
   <package id="jQuery" version="3.3.1" targetFramework="net472" />
   <package id="jQuery.Validation" version="1.17.0" targetFramework="net472" />
-  <package id="Microsoft.AspNet.Mvc" version="5.2.7" targetFramework="net472" />
-  <package id="Microsoft.AspNet.Razor" version="3.2.7" targetFramework="net472" />
-  <package id="Microsoft.AspNet.TelemetryCorrelation" version="1.0.5" targetFramework="net472" />
+  <package id="Microsoft.AspNet.Mvc" version="5.2.4" targetFramework="net472" />
```
[the remark this answers is not in the document]
Comment (jasonjoh) [reply Demos/01-create-app/graph-tutorial/packages.config]: Per discussion this was output from VS template.

File: tutorial/add-ms-graph.md [hunk 2]
```diff
@@ -89,6 +91,15 @@ namespace graph_tutorial.Controllers
         public async Task<ActionResult> Index()
         {
             var events = await GraphHelper.GetEventsAsync();
+
+            foreach (var ev in events)
```
[the remark this answers is not in the document]
Comment (rob-windsor) [reply tutorial/add-ms-graph.md]: @jasonjoh Do you want the comment to explain the conversion from UTC to local time? If not please give me an idea what you are looking for.

**Comment by rob-windsor** [comment]
@jasonjoh I'm going to summarize the changes I'll be making just to ensure we are on the same page.

**Port number**
I won't be making any changes to the tutorial instructions but I will ensure that the completed projects (i.e. the ones in the Demos folder) use port 64107.

**Package versions**
I won't make any changes to the versions of packages installed by the Visual Studio project template. For example, the project will use  Microsoft.AspNet.Mvc v5.2.4 rather than v5.2.7.  In the tutorial I will update the instructions to explicitly install Microsoft.Identity.Client v3.0.8 and Microsoft.Graph v1.15.0.

**Comment in the Index action of the CalendarController**
I am going to add a code comment to the foreach loop explaining what it is doing. I am waiting on the answer to my question above to confirm what you want to comment to say. I assume that it will be related to converting the start of end times of each event from UTC to local time.

Please let me know if I have missed or misunderstood anything.

File: tutorial/add-ms-graph.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jasonjoh) [reply tutorial/add-ms-graph.md]: Yep basically. When I first was looking at this I was thinking, what's this for? I had to look at it a couple of times to realize you were modifying the collection in place, then it all clicked.

**Comment by jasonjoh** [comment]
That's fine Rob. Thanks for doing this.

File: tutorial/add-ms-graph.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewconnell) [reply tutorial/add-ms-graph.md]: Resolved in last batch of comments below

**Comment by andrewconnell** [comment]
@jasonjoh addressed all feedback in the last batch of commits

**Pull request merged**